numpyro/contrib/control_flow/cond.py

```diff
@@ -0,0 +1,131 @@
+            if rng_key is not None
+            else subs_map
+        )
+        return subs_map(site)
+    return None
+
+
+def wrap_fn(fn, substitute_stack):
+    def wrapper(wrapped_operand):
+        rng_key, operand = wrapped_operand
+
+        with handlers.block():
+            seeded_fn = handlers.seed(fn, rng_key) if rng_key is not None else fn
+            for subs_type, subs_map in substitute_stack:
+                subs_fn = partial(_subs_wrapper, subs_map)
+                if subs_type == "condition":
+                    seeded_fn = handlers.condition(seeded_fn, condition_fn=subs_fn)
+                elif subs_type == "substitute":
+                    seeded_fn = handlers.substitute(seeded_fn, substitute_fn=subs_fn)
+
+            with handlers.trace() as trace:
+                value = seeded_fn(operand)
+
+        return value, PytreeTrace(trace)
+
+    return wrapper
+
+
+def cond_wrapper(
+    pred, true_fun, false_fun, operand, rng_key=None, substitute_stack=None
+):
+    if substitute_stack is None:
+        substitute_stack = []
+
+    wrapped_true_fun = wrap_fn(true_fun, substitute_stack)
+    wrapped_false_fun = wrap_fn(false_fun, substitute_stack)
+    wrapped_operand = device_put((rng_key, operand))
+    return lax.cond(pred, wrapped_true_fun, wrapped_false_fun, wrapped_operand)
+
+
+def cond(pred, true_fun, false_fun, operand):
+    """
+    This primitive conditionally applies `true_fun` or `false_fun`. See
+    :func:`jax.lax.cond` for more information.
+
+    **Usage**:
+
+    .. doctest::
+
+       >>> import numpyro
+       >>> import numpyro.distributions as dist
+       >>> from numpyro.contrib.control_flow import cond
+       >>>
+       >>> def model():
+       ...     def true_fun(_):
+       ...         numpyro.sample("x", dist.Normal(20.0))
+       ...
+       ...     def false_fun(_):
+       ...         numpyro.sample("x", dist.Normal(0.0))
+       ...
+       ...     cluster = numpyro.sample("cluster", dist.Normal())
+       ...     cond(cluster > 0, true_fun, false_fun, None)
+       >>>
+       >>> def guide():
+       ...     m1 = numpyro.param("m1", 10.0)
+       ...     s1 = numpyro.param("s1", 0.1, constraint=dist.constraints.positive)
+       ...     m2 = numpyro.param("m2", 10.0)
+       ...     s2 = numpyro.param("s2", 0.1, constraint=dist.constraints.positive)
+       ...
+       ...     def true_fun(_):
+       ...         numpyro.sample("x", dist.Normal(m1, s1))
+       ...
+       ...     def false_fun(_):
+       ...         numpyro.sample("x", dist.Normal(m2, s2))
+       ...
+       ...     cluster = numpyro.sample("cluster", dist.Normal())
+       ...     cond(cluster > 0, true_fun, false_fun, None)
+       >>>
+       >>> svi = SVI(model, guide, numpyro.optim.Adam(1e-2), Trace_ELBO(num_particles=100))
+       >>> params, losses = svi.run(random.PRNGKey(0), num_steps=2500)
+
+        .. warning:: This is an experimental utility function that allows users to use
```

Comment:
I think you need to remove 1 block of indentation here and below.

Reply:
Good catch, thanks! Have checked this locally now and I think it's fixed.